src/utils/management/commands/install_janeway.py

```diff
@@ -49,9 +51,19 @@ def handle(self, *args, **options):
                 press.main_contact = input('Press main contact (email): ')
                 press.save()
 
-            print("Thanks! We will now set up out first journal.\n")
+            logger.info("Thanks! We will now set up out first journal.\n")
             journal = journal_models.Journal()
-            journal.code = input('Journal #1 code: ')
+
+            while True:
+                code = input('Journal #1 code (15 characters max): ')
+
+                if len(code) <= 15:
+                    journal.code = code
+                    journal.save()
+                    break
+                else:
+                    logger.warning('Jouranl code must be 15 characters or less.')
```

Comment:
I see your point about the validation through the model within the transaction. Perhaps we should use a form instead and validate all the fields instead of hardcoding the logic here?

Reply:
It is possible to fetch the field length with `Journal._meta.get_field('code').max_length`

Thoughts?

Reply:
Its another option but only covers one form of validation for a single field, so we might have to add all the others. If we use a form then we can use the standard validation methods from django